doc/observables/registry_keyval.md

```diff
@@ -0,0 +1,103 @@
+
+## Introduction
+
+This guide will document how to combine Registry Key and Registry Value Entries into a single canonical string format which can be used as an observable value in CTIM.
+
+### Registry Keys
+
+The Windows Registry is a file database that allows multiple **Value Entries** to be stored under a **Key Path**.  The registry supports seven root keys, some of which have strictly controlled sets of officially designated subkeys, into which additional subkeys can be nested.
+
+#### Registry Key Name Constraints
+
+> Each key has a name consisting of one or more printable characters. Key names are not case sensitive. Key names cannot include the backslash character (), but any other printable character can be used. Value names and data can include the backslash character.
+
+Thus, no printable character, except for the backslash (`\`), can reliably and safely be used as a delimiter in key strings.
+
+### Registry Value Entries
+
+Each subkey in the registry can have zero or more **Value Entries** stored under it.
+
+Every **value entry** is composed of exactly three parts:
+
+1. The `name` of the value, also known as the `value_name`.
+2. The `type` of the value, also known as the `value_type`.
+3. The `value` itself, also known as the `value_content`.
+
+The canonical representation of the Registry Value Entry is `"value_name" = value_type:value_content`
+
+### Existing Representations
+
+#### Example 01: Windows Registry Export Format
+
+The standard representation of Registry data is to supply the key path, and then the value information beneath it, like so:
+
+```
+[HKEY_LOCAL_MACHINE\SOFTWARE\Foobar]
+"Value A"="<String value data with escape characters>"
+"Value B"=hex:<Binary data (as comma-delimited list of hexadecimal values)>
+"Value C"=dword:<DWORD value integer>
+```
+
+
+#### Example 02: Registry Key-Value Pair From the TALOS Blog:
+
+```
+    <HKCU>\SOFTWARE\MICROSOFT\WINDOWS\CURRENTVERSION\RUN
+        Value Name: api-PQEC
+```
+
+
+### Goal
+
+There does not exist a standard, ubiquitous method of combining a registry key path and a value entry into a single string format that we could use for an observable string.  So, we have determined to define our own canonical string representation that combines the entire key path, as well as the information about the value entry.
+
+This will provide us with something very similar to the [socket address](https://en.wikipedia.org/wiki/Network_socket#Socket_addresses), which is the canonical representation of an IP address and Port combination.  In a single string, we will be able to safely and reliably designate both the key path in its entirety, as well as the value name, value type, and value content.
+
+
+
+## Canonical String Format
+
+We can encode the registry key and value entry using the following canonical form:
+
+```
+registry_kv: "HKEY_CURRENT_USER\SOFTWARE\MICROSOFT\WINDOWS\CURRENTVERSION\RUN\\\"api-PQEC\"=REG_DWORD:0x800000"
+```
```

Comment:
Maybe it's time to change the structure of the `Observable` value object by allowing more than a `type` and a `value`. [STIX 2](http://docs.oasis-open.org/cti/stix/v2.0/cs01/part4-cyber-observable-objects/stix-v2.0-cs01-part4-cyber-observable-objects.html#_Toc496716290) encodes Windows Registry keys in a more structured way. I prefer their approach, it's more readable by a human or a machine.

Reply:
I had considered using combinations of Observables, but not adding optional fields  to the Observable value object.  I'm in favor of holding off on this until @craigbro has a chance to weigh in.